dataprofiler/profilers/profile_builder.py

```diff
@@ -321,20 +322,17 @@ def get_base_props_and_clean_null_params(self, df_series, sample_size,
                     # need to make new var
                     row_value = str(df_series_subset[row])
                     na_columns.setdefault(row_value, list()).append(row)
-                    
-            # Drop the values that matched regex_na
-            df_series_subset = df_series_subset[~matching_na_elements]
-            
-            true_sample_list += df_series_subset.index.tolist()
+                else:
+                    true_sample_list.add(row)
```

Comment:
This change shaves off ~5% overall runtime:

**Current** (in master):
```
         39712807 function calls (39573098 primitive calls) in 58.290 seconds

   Ordered by: internal time
   List reduced from 9663 to 50 due to restriction <50>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     1304   19.516    0.015   19.516    0.015 {method 'acquire' of '_thread.lock' objects}
  496/478    9.761    0.020    9.761    0.020 {built-in method numpy.array}
 16716568    3.961    0.000    3.961    0.000 {method 'search' of 're.Pattern' objects}
     8375    3.960    0.000    3.961    0.000 {built-in method builtins.sorted}
        6    3.231    0.538   29.630    4.938 profile_builder.py:254(get_base_props_and_clean_null_params)
```

**New in PR**
```
         56424765 function calls (56285092 primitive calls) in 55.109 seconds

   Ordered by: internal time
   List reduced from 9607 to 50 due to restriction <50>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     1304   19.222    0.015   19.222    0.015 {method 'acquire' of '_thread.lock' objects}
  412/406    6.258    0.015    6.258    0.015 {built-in method numpy.array}
        6    4.642    0.774   26.368    4.395 profile_builder.py:254(get_base_props_and_clean_null_params)
```